### PR TITLE
per naming convention and #4921 this should make this `Rectangle.validate` private.

### PR DESCRIPTION
**Pull request by hotpocket**
# Description
The `Rectangle.validate` function is only called in pragmas and has thus been an empty function call.  Per issue #4921 it was suggested to make this private, presumably so that this behavior is explicit. 

## Testing plan
All tests pass that were previously passing.  There was one test that failed on my Ubuntu 22.04.05 OS.  This is listed below:
```
TOTAL: 1 FAILED, 14300 SUCCESS

1) renders a point
     Scene/Vector3DTilePoints
     Expected to render [255,255,255,255], but actually rendered [254,254,254,255].
    at <Jasmine>
    at packages/engine/Specs/Scene/Vector3DTilePointsSpec.js:177:25 <- Build/Specs/SpecList.js:236887:24
```
This test was failing on unmodified version of main with the command `npm run test`. I assume this is "ok" with respect to this issue. 

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by ggetz** [comment]
@jjspace Could you please give this a review?

File: packages/engine/Source/Core/Rectangle.js [hunk 1]
```diff
@@ -542,7 +542,7 @@ Rectangle.prototype.equalsEpsilon = function (other, epsilon) {
  * @exception {DeveloperError} <code>east</code> must be in the interval [<code>-Pi</code>, <code>Pi</code>].
  * @exception {DeveloperError} <code>west</code> must be in the interval [<code>-Pi</code>, <code>Pi</code>].
  */
-Rectangle.validate = function (rectangle) {
+Rectangle._validate = function (rectangle) {
```
Comment (jjspace) [inline packages/engine/Source/Core/Rectangle.js:545]: As we discussed last night we should go through the deprecation process for this function. You can read about it in our [Coding Guide](https://github.com/CesiumGS/cesium/blob/main/Documentation/Contributors/CodingGuide/README.md#deprecation-and-breaking-changes). Basically rename this function but also add the `Rectangle.validate` function back which just calls through to this one and has the `@deprecated` tag and `deprecatedWarning`.

**Review by jjspace — CHANGES_REQUESTED** [review]
This is looking good, just a few more steps to make sure changes to the public API are make responsibly.

**Comment by hotpocket** [comment]
@jjspace I believe this is ready.

**Review by jjspace — APPROVED** [review]
Looks good to me now, thanks @hotpocket!

**Review by jjspace — APPROVED** [review]
Sorry for any noise this caused in your notifications @hotpocket, we just [updated prettier](https://github.com/CesiumGS/cesium/pull/12206) and I was trying to update your branch for you but turned out a little trickier than I thought but that was on my side.

This is finally good to go now

**Review by lukemckinstry — APPROVED** [review]
Approving following @jjspace's Prettier updates

**Comment by hotpocket** [comment]
> Sorry for any noise this caused in your notifications @hotpocket, we just [updated prettier](https://github.com/CesiumGS/cesium/pull/12206) and I was trying to update your branch for you but turned out a little trickier than I thought but that was on my side.
> 
> This is finally good to go now

I am comforted to know that I am not the only one that still fights with git :slightly_smiling_face: ([xkcd](https://xkcd.com/1597/)).  Thanks @jjspace

**Pull request merged**